### ISSUE TITLE
Feature: Improve elevation of Bash commands

### ISSUE THREAD

**Issue by dennis-si** [gerardog/gsudo#56]
firstly I wanted to use dd under windows 10, since it's not possible under wsl2 environment
Discovered Git Bash for windows has dd tool, already had Git Bash installed, great then added it to Windows Terminal.
But could only git to run correctly if I started Terminal shell with "run as administrator", otherwise permission denied.
Found Gsudo, was impressed and went to use it with dd with the following
$ gsudo dd if=/dev/zero of=/dev/sdc bs=1M count=10
/usr/bin/dd: failed to open '/Device/Null': No such file or directory
Info: Process exited with code 1

Any ideas would be greatly appreciated. :-)
Cheers, Dennis 

btw, sudo bash & then dd if=/dev/zero of=/dev/sdc bs=1M count=10, works also

**Comment by gerardog** [comment]
Ok. Let's split this issue in two parts:

First, you should open git-bash as admin the regular way (right-click, run as admin), to test if `dd` works for you. gsudo can help you elevate, but even if the elevation worked, I [cannot assure](https://superuser.com/questions/839502/windows-equivalent-for-dd) that `dd` will work the way you want. 

Second, regarding using gsudo from git-bash: Right now gsudo does not have special logic for 'git-bash` as a shell, so it behaves as-if it were invoked from cmd. It tries to execute the command as a windows command (as-in with 'cmd.exe /c' prefix).

For example: 
```
$ gsudo --debug echo Hello World!
Debug: Command to run: C:\WINDOWS\system32\cmd.exe /c echo Hello World!

Hello World!
```
In this case the `cmd echo` is equivalent to `bash echo` so the user does not notice, but it is (right now) running an elevated cmd command, not bash. This is the first thing to fix.

Then there is at least one more issue to address: As `gsudo.exe` is a windows app, `git-bash` translates all parameters from unix-paths to windows-paths making things worse:

```
$ gsudo echo /etc/hosts
"C:/Program Files/Git/etc/hosts"
```
Not what we wanted. Another example:

```
$ gsudo echo dd if=/dev/zero of=/dev/sdc bs=1M count=10
dd if=/Device/Null "of=C:/Program Files/Git/dev/sdc" bs=1M count=10
```
The path translation ocurrs before even invoking gsudo and totally breaks the command in a way that gsudo can't reverse. 

So, this needs to be addressed in a similar way as of #55: Creating a "gsudo-wrapper bash script", that prevents unwanted path translation, and invokes gsudo with special considerations.

The bash script needs some work and proper testing. As a starting point, running `$ gsudo.exe bash` elevates the bash in the same window, and `$ gsudo.exe bash -c "echo 123 /dev/null"` prevents path-translation. 

On both git-bash and WSL, when runs `$ gsudo` and there is a `gsudo` bash script in the path, that one is run instead of `gsudo.exe`. So this wrapper must address both scenarios.

**Comment by gerardog** [comment]
Implemented in [v1.0.0](https://github.com/gerardog/gsudo/releases/tag/v1.0.0) (PR #94)
